simulator/memory/argv_loader/argv_loader.h

```diff
@@ -0,0 +1,39 @@
+/**
+ * argv_loader.h: loader of arguments to guest memory
+ * @author Rustem Yunusov <yunusov.rr@phystech.edu>
+ * Copyright 2019 MIPT-MIPS iLab project
+ */
+
+#ifndef ARGV_LOADER_H
+#define ARGV_LOADER_H
+
+#include <infra/argv.h>
+#include <infra/byte.h>
+#include <infra/types.h>
+#include <memory/memory.h>
+
+class ArgvLoader
+{
+public:
+    explicit ArgvLoader( const char* const* argv, const char* const* envp = nullptr);
+    ~ArgvLoader();
+
+    size_t load_argv_to( const std::shared_ptr<FuncMemory>& mem, Addr addr);
+private:
+    const int argc;
+    const char* const* argv;
+    const char* const* envp;
+    Addr offset;
+    Addr envp_offset;
+
+    size_t place_nullptr( const std::shared_ptr<FuncMemory>& plain_mem, Addr addr)
+    {
+        const char* null = nullptr;
+        return plain_mem -> memcpy_host_to_guest( addr, byte_cast( &null), bytewidth<char*>);
```

Comment:
Again, you are using host pointer while you have to use guest pointer:
```c++
Addr null = 0;
return plain_mem->write<Addr>( addr, null); // please check the order of arguments
```

Reply:
If I load argv [ i ], which is 'const char*', what should I use in write(): 'Addr' or 'const char*'?
I suppose 'const char*'.
But null is of the same type as argv [ i ], that's why I'm confused.

Reply:
You must not write `argv[i]` to guest memory, since it is a **host** memory pointer. You have to write **guest** memory pointer here, i.e. the **guest** memory address (Addr) where the `*argv[i]` data resides in guest memory.